### ISSUE TITLE
Method ambiguity detection: false positive?

### ISSUE THREAD

**Issue by pygy** [JuliaLang/julia#6048]
```Julia
foo(b::Bar...) = 1
# foo (generic function with 1 methods)

foo(a::Any, b::Bar) = 2
# foo (generic function with 2 methods)

foo(b::Bar, a::Any) = 3
# Warning: New definition
#     foo(Bar,Any)
# is ambiguous with:
#     foo(Any,Bar)
# To fix, define
#     foo(Bar,Bar)
# before the new definition.
# foo (generic function with 3 methods)
```

Semantically, foo(Bar,Bar)` is covered by the first definition, and the above example behaves as expected.

Is the warning really warranted?

**Comment by timholy** [comment]
Closed by #16125 or #16220

**Comment by pygy** [comment]
Indeed, many thanks :-)
